src/pages/suggestions/index.tsx

```diff
@@ -5,50 +5,70 @@ import { PlusIcon } from "@heroicons/react/24/solid"
+  const [showSpinner, setShowSpinner] = useState<boolean>(true)
+  const [showLoadingError, setShowLoadingError] = useState<boolean>(false)
+  const [noSuggestionsMade, setNoSuggestionsMade] = useState<boolean>(false)
 
   useEffect(() => {
-    const fetchSuggestions = async () => {
-      const { data, error } = await getSuggestions(supabaseClient)
-      if (error) throw error
-      if (data) {
-        const dataSuggestions = data as Suggestion[]
-        setSuggestions(dataSuggestions)
-      }
-      setIsLoading(false)
-    }
-
-    // TODO Implement error handling
-    fetchSuggestions().catch(console.error)
+    getSuggestions(supabaseClient)
+      .then((response) => {
+        response.data ? setSuggestions(response.data as Suggestion[]) : setShowLoadingError(true)
+      })
+      .catch(() => setShowLoadingError(true))
+      .finally(() => {
+        setShowSpinner(false)
+      })
   }, [supabaseClient])
 
-  // TODO Show loader when loading
-  if (isLoading) {
-    return <p>Loading...</p>
-  }
+  useEffect(() => {
+    setNoSuggestionsMade(suggestions.length === 0)
+  }, [suggestions])
 
   return (
     <div className={"page-wrapper"}>
-      <div className={"flex justify-between"}>
-        <div className={"page-header"}>Suggestions</div>
-        <PlusIcon
-          className={"h-8 w-8 cursor-pointer text-black hover:text-zinc-400"}
-          onClick={() => router.push("/suggestions/new")}
-        />
+      <div className={"flex justify-between pb-6"}>
+        <div className={"text-2xl font-semibold leading-8"}>Suggestions</div>
+        <div>
+          <PlusIcon className={"h-8 w-8 text-black"} onClick={() => {}} />
+        </div>
       </div>
 
-      <div className={"flex flex-wrap justify-center gap-6"}>
-        {/* TODO If no suggestions are found, show that to the user */}
-        {suggestions?.map((suggestion) => (
-          <SuggestionCard key={suggestion.id} suggestion={suggestion} />
-        ))}
-      </div>
+      {showSpinner ? (
+        <div className={"text-center"} data-cy="suggestions-spinner">
+          <Spinner size={10} />
+        </div>
+      ) : (
+        <div className={"flex flex-wrap justify-center gap-6"} data-cy="suggestions-list">
+          {suggestions.map((suggestion) => (
+            <SuggestionCard key={suggestion.id} suggestion={suggestion} />
+          ))}
+        </div>
+      )}
+
+      {showLoadingError && (
+        <div className={"mt-6"}>
+          <ErrorPopup
+            text={"Failed to load suggestions."}
+            closePopup={() => setShowLoadingError(false)}
+          />
+        </div>
+      )}
+
+      {noSuggestionsMade && (
+        <div className={"max-w-m flex items-center justify-center gap-4 text-zinc-400"}>
+          <div>
+            <MagnifyingGlassCircleIcon className={"h-[50px] w-[50px]"} />
+          </div>
+          <p>Looks like there are no suggestions made yet! Feel free to start adding them.</p>
+        </div>
+      )}
```

Comment:
This is always shown when it is loading the suggestions. I think you want to hide it then as well.